x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/schedule/utils/convert_form_data.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IUiSettingsClient } from '@kbn/core/public';
+import { getEsQueryConfig } from '@kbn/data-plugin/common';
+import type { AIConnector } from '@kbn/elastic-assistant';
+import type { DataViewBase } from '@kbn/es-query';
+
+import { convertToBuildEsQuery } from '../../../../../common/lib/kuery';
+import type { AttackDiscoveryScheduleSchema } from '../edit_form/types';
+import { getGenAiConfig } from '../../../use_attack_discovery/helpers';
+import { parseFilterQuery } from '../../parse_filter_query';
+
+export const convertFormDataInBaseSchedule = (
+  scheduleData: AttackDiscoveryScheduleSchema,
+  alertsIndexPattern: string,
+  connector: AIConnector,
+  indexPattern: DataViewBase,
+  uiSettings: IUiSettingsClient
+) => {
+  const alertsSelectionSettings = scheduleData.alertsSelectionSettings;
+
+  const [filterQuery, kqlError] = convertToBuildEsQuery({
+    config: getEsQueryConfig(uiSettings),
+    indexPattern,
```

Comment:
Similarly to https://github.com/elastic/kibana/pull/205137#discussion_r1896745964

In `main`, the `convertToBuildEsQuery` function called here accepts a parameter named `dataViewSpec`, which is typed as:

```typescript
dataViewSpec: DataViewSpec | undefined;
```

In `8.x`, the `convertToBuildEsQuery` function takes a different parameter named `indexPattern` instead of `dataViewSpec`, and it's typed differently:

```typescript
indexPattern: DataViewBase | undefined;
```

The page was updated to call the `convertToBuildEsQuery` function using the `8.x` version of the parameter.